java/src/jmri/util/MathUtil.java

```diff
@@ -1090,6 +1089,21 @@ public static Rectangle2D rectangleAtPoint(@Nonnull Point2D p, Double width, Dou
         return new Rectangle2D.Double(p.getX(), p.getY(), width, height);
     }
 
+    /**
+     * reverse an array of Point2D's
+     *
+     * @param points the array
+     * @return the reversed array
+     */
+    public static Point2D[] reverse(Point2D[] points) {
```

Comment:
Just a suggestion, not a formal request for changes, but this seems well suited to use of generics.  That would make it be a routine for reversing an array of _any_ type.
```
public static <T> void reverse(T[] points) {
    T[] results = new T[points.length];
    List<T> itemList = Arrays.asList(points);
```
et al.

Reply:
I'll try to follow your suggestion. Thanks for the review!

Reply:
Java does not allow the array to be generic:
```
    [javac] Compiling 1 source file to C:\Users\Klaus\JMRI-sourcecurrent\target\classes
    [javac] C:\Users\Klaus\JMRI-sourcecurrent\java\src\jmri\util\MathUtil.java:1100: error: generic array creation
    [javac]         T[] results = new T[points.length];
    [javac]                       ^
    [javac] 1 error
```

This [tutorial](https://www.softwaretestinghelp.com/java-generic-array/) suggests to simulate generic arrays.
But it is not straight forward for me. Would it be okay to leave it as it is?

Reply:
> Would it be okay to leave it as it is?

Thanks for trying it. Sorry for the confusion on my part.  Yes, of course it's fine!

Reply:
I played with this a little.  This works nicely:
```
    public static <T> T[] reverse(T[] points) {
        T[] results = java.util.Arrays.copyOf(points, points.length);
        for (int i = 0; i < points.length; i++) {
            results[i] = points[points.length-i-1];
        }
        return results;
    }
```
No need to change it, I was just curious about how this could be done.

Reply:
Of course I tried your solution. 
At runtime I get a ```java.lang.NoSuchMethodError: 'java.awt.geom.Point2D[] jmri.util.MathUtil.reverse(java.awt.geom.Point2D[])'```. Perhaps I have to change the calling part?

Reply:
The new reverse method is now in use.